### PR TITLE
CLI code refactor

### LINKED ISSUE #658: use require() in index.js to source from /src/cli/
The `index.js` file in the root directory runs the CLI interface. But the CLI code is a bit long now, and could be easier to read. Let's break it up into a few different files kept in `/src/cli/` and use `require()` to access them from `index.js`. This will result in cleaner and easier to read code. 

Personally I like this type of task! It can be pretty fun to tidy things up. The CLI tests should help us ensure this is done properly, too. We'd love some help!

### PR DESCRIPTION
**Pull request by VibhorCodecianGupta**
Fixes #658 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
There's a let based error in node 4, however. I think there's an issue open but we should be able to set strict mode and resolve it. What do you think?

Great work!!!

**Comment by VibhorCodecianGupta** [comment]
@jywarren using strict is always good, let's do it!
However, is the let based error affecting this PR too? I don't think it is..

**Comment by HarshKhandeparkar** [comment]
@jywarren we can compile the CLI code using [NectarJS](https://github.com/NectarJS/nectarjs/blob/master/README.md)

**Comment by VibhorCodecianGupta** [comment]
@jywarren is this a good enough block to be merged?

File: index.js [hunk 1]
```diff
@@ -49,12 +38,12 @@ if (program.saveSequence) {
...
 
   // User must input an image.
-  require("fs").access(program.image, function(err) {
+  fs.access(program.image, function(err) {
     if (err) exit("Can't read file.");
   });
 
   // User must input a step. If steps exist, check that every step is a valid step.
-  if (!program.step || !validateSteps(program.step))
+  if (!program.step || !(require('./src/CliUtils').validateSteps(program.step, sequencer)))
```
Comment (HarshKhandeparkar) [inline index.js]: How about requiring everything at the start of the file for consistency?

File: index.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
 #!/usr/bin/env node
 
-require("./src/ImageSequencer");
+require('./src/ImageSequencer');
 sequencer = ImageSequencer({ ui: false });
-var Spinner = require("ora");
+var fs = require('fs')
+var program = require('commander');
```
Comment (HarshKhandeparkar) [inline index.js]: You can use a comma instead of using var keyword every line.

**Comment by jywarren** [comment]
Hmm, is Travis stuck?

**Comment by VibhorCodecianGupta** [comment]
@Mridul97 @jywarren Travis seems to be stuck. Any ideas?

**Comment by HarshKhandeparkar** [comment]
@VibhorCodecianGupta try running the tests locally and see if anything pops up

**Comment by HarshKhandeparkar** [comment]
Maybe infinite recursion of require ?

**Comment by HarshKhandeparkar** [comment]
Did you try using the new CLI code locally? See if it gets stuck

**Comment by VibhorCodecianGupta** [comment]
@HarshKhandeparkar checked and rechecked, everything's fine. #650 and #696 face similar issues

**Comment by HarshKhandeparkar** [comment]
Is that because node 4 support was dropped? Maybe the travis file is broken somehow

**Comment by HarshKhandeparkar** [comment]
Is your forked travis file counted? Or only the upstream file is counted?

**Comment by HarshKhandeparkar** [comment]
Oh the progress can be seen. It seems like it is stuck at upgrading system packages of some sort.

**Comment by VibhorCodecianGupta** [comment]
@jywarren @HarshKhandeparkar Travis was just behaving weirdly I guess? The last two commits are just do/undo commits and the job kickstarted again. Travis passes now

**Comment by HarshKhandeparkar** [comment]
Yes I saw that. Maybe it was a travis bug of some sort.

**Comment by HarshKhandeparkar** [comment]
https://github.com/publiclab/image-sequencer/pull/696 is still in progress though. Maybe any commit would reset the procedure and fix this?

**Comment by VibhorCodecianGupta** [comment]
@HarshKhandeparkar I guess that's it. Once the job kickstarts again, Travis resumes.
Also, could you review the changes?

**Comment by HarshKhandeparkar** [comment]
No need of a GIF. Thanks! Looks good!

**Comment by HarshKhandeparkar** [comment]
> I'm on a slow internet here 

Same here. I'm using BSNL so no hopes for the next 💯 years. The latency is more than what it would take to compile image-sequencer.

**Comment by VibhorCodecianGupta** [comment]
@jywarren I think this is a complete blob of work. Thoughts?

**Comment by jywarren** [comment]
Looks good! Merging now, and great work, again!

**Pull request merged**